Framework/Algorithms/src/DiffractionFocussing2.cpp

```diff
@@ -224,7 +224,16 @@ void DiffractionFocussing2::exec() {
       // get histogram version of the data
       if (eventInputWS) {
         const EventList &el = eventInputWS->getSpectrum(inWorkspaceIndex);
-        el.generateHistogram(group2xstep.at(group), Xout.rawData(), Yout, Eout);
+        // generateHistogram overwrites the data in Y and E so write to a temporary vector
+        MantidVec Ytemp;
+        MantidVec Etemp;
+        el.generateHistogram(group2xstep.at(group), Xout.rawData(), Ytemp, Etemp);
+        // accumulate the histogram into the output
+        std::transform(Ytemp.cbegin(), Ytemp.cend(), Yout.begin(), Yout.begin(),
```

Comment:
I suppose this was causing issue before. Becaue Xout, Yout and Eout was overwritten. Now the data from generateHistorgram gets added into Yout and Eout.